tests/test-class-wpseo-breadcrumbs.php

```diff
@@ -50,4 +61,39 @@ public function test_breadcrumb_after() {
 		// @todo Test actual breadcrumb output.
 	}
 
+	/**
+	 * Tests getting the url for a private post.
+	 *
+	 * @covers WPSEO_Breadcrumbs::get_link_url_for_id()
+	 */
+	public function test_getting_url_of_private_post() {
+		$breadcrumbs = new WPSEO_Breadcrumbs_Double();
+
+		$post = $this->factory()->post->create_and_get( array( 'post_status' => 'private' ));
+		$this->assertEquals( '', $breadcrumbs->get_link_url_for_id( $post->ID ) );
+	}
+
+	/**
+	 * Tests getting the url for a public post.
+	 *
+	 * @covers WPSEO_Breadcrumbs::get_link_url_for_id()
+	 */
+	public function test_getting_url_of_public_post() {
+		$breadcrumbs = new WPSEO_Breadcrumbs_Double();
+
+		$post = $this->factory()->post->create_and_get();
+		$this->assertEquals( get_permalink( $post->ID ), $breadcrumbs->get_link_url_for_id( $post->ID ) );
+	}
+
+	/**
+	 * Tests getting the url for a non existing post id.
+	 *
+	 * @covers WPSEO_Breadcrumbs::get_link_url_for_id()
+	 */
+	public function test_getting_url_of_a_non_existing_post() {
+		$breadcrumbs = new WPSEO_Breadcrumbs_Double();
+
+		$this->assertEquals( '', $breadcrumbs->get_link_url_for_id( 500000 ) );
```

Comment:
It would be more secure to test against a value of `0` as that post ID should never exist.